### ISSUE TITLE
Old Regression, KeyError: "'__name__' not in globals"

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [nelfin/pylint-protobuf#51]
# Summary
Adding pylint protobuf to my pylint configuration has difficulty with a couple files that causes it to just error out and fail. The smallest example I've found is just importing a tensorflow_transform module. This one line file,

`
from tensorflow_transform.beam import tft_unit
`

fails. The file checks correctly back in 0.13 and then fails for 0.14 so this is an old regression.

# Details
The error message is,

```
Exception on node <ImportFrom l.1 at 0x10eff1350> in file '/Users/mdrissi/Snapchat/Dev/pylint_issues/pylint_protobuf_pain.py'
Traceback (most recent call last):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/__init__.py", line 24, in run_pylint
    PylintRun(sys.argv[1:])
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/lint/run.py", line 384, in __init__
    linter.check(args)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 974, in check
    self.get_ast, self._iterate_file_descrs(files_or_modules)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1007, in _check_files
    self._check_file(get_ast, check_astroid_module, name, filepath, modname)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1033, in _check_file
    check_astroid_module(ast_node)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1171, in check_astroid_module
    ast_node, walker, rawcheckers, tokencheckers
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/lint/pylinter.py", line 1215, in _check_astroid_module
    walker.walk(ast_node)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/utils/ast_walker.py", line 77, in walk
    self.walk(child)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/utils/ast_walker.py", line 74, in walk
    callback(astroid)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1212, in visit_importfrom
    self._check_module_attrs(node, module, name.split("."))
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint/checkers/variables.py", line 1943, in _check_module_attrs
    module = next(module.getattr(name)[0].infer())
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 593, in getattr
    result = [self.import_module(name, relative_only=True)]
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 686, in import_module
    return AstroidManager().ast_from_module_name(absmodname)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/manager.py", line 205, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/manager.py", line 116, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/builder.py", line 135, in file_build
    return self._post_build(module, encoding)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/builder.py", line 155, in _post_build
    self.delayed_assattr(delayed)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/builder.py", line 225, in delayed_assattr
    for inferred in node.expr.infer():
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 353, in infer
    yield from self._infer(context, **kwargs)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/bases.py", line 144, in _infer_stmts
    for inferred in stmt.infer(context=context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 97, in wrapped
    generator = _func(node, context, **kwargs)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 860, in infer_assign
    stmts = list(self.assigned_stmts(context=context))
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/protocols.py", line 330, in _arguments_infer_argname
    is_metaclass = isinstance(cls, nodes.ClassDef) and cls.type == "metaclass"
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 1879, in _class_type
    if _is_metaclass(klass):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 1848, in _is_metaclass
    for baseobj in base.infer():
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 353, in infer
    yield from self._infer(context, **kwargs)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 326, in infer_attribute
    yield from owner.igetattr(self.attrname, context)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/bases.py", line 144, in _infer_stmts
    for inferred in stmt.infer(context=context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 97, in wrapped
    generator = _func(node, context, **kwargs)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 279, in infer_import_from
    module = self.do_import_module()
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/mixins.py", line 102, in do_import_module
    modname, level=level, relative_only=level and level >= 1
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 686, in import_module
    return AstroidManager().ast_from_module_name(absmodname)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/manager.py", line 205, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/manager.py", line 116, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/builder.py", line 135, in file_build
    return self._post_build(module, encoding)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/builder.py", line 155, in _post_build
    self.delayed_assattr(delayed)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/builder.py", line 225, in delayed_assattr
    for inferred in node.expr.infer():
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 353, in infer
    yield from self._infer(context, **kwargs)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 374, in infer_subscript
    for value in self.value.infer(context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 299, in infer_attribute
    for owner in self.expr.infer(context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 299, in infer_attribute
    for owner in self.expr.infer(context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/bases.py", line 144, in _infer_stmts
    for inferred in stmt.infer(context=context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/bases.py", line 144, in _infer_stmts
    for inferred in stmt.infer(context=context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 233, in infer_call
    for callee in self.func.infer(context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 299, in infer_attribute
    for owner in self.expr.infer(context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/bases.py", line 144, in _infer_stmts
    for inferred in stmt.infer(context=context):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/decorators.py", line 97, in wrapped
    generator = _func(node, context, **kwargs)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/inference.py", line 286, in infer_import_from
    stmts = module.getattr(name, ignore_locals=module is self.root())
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 593, in getattr
    result = [self.import_module(name, relative_only=True)]
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/scoped_nodes.py", line 686, in import_module
    return AstroidManager().ast_from_module_name(absmodname)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/manager.py", line 205, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/manager.py", line 116, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/builder.py", line 135, in file_build
    return self._post_build(module, encoding)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/builder.py", line 159, in _post_build
    module = self._manager.visit_transforms(module)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/manager.py", line 93, in visit_transforms
    return self._transform.visit(node)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/transforms.py", line 95, in visit
    return self._transform(module)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/astroid/transforms.py", line 44, in _transform
    ret = transform_func(node)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint_protobuf/transform.py", line 566, in transform_module
    cls = mod_node_to_class(mod, name)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint_protobuf/transform.py", line 545, in mod_node_to_class
    ns = _exec_module(mod)
  File "/Users/mdrissi/.virtual_envs/debug_pylint/lib/python3.7/site-packages/pylint_protobuf/transform.py", line 539, in _exec_module
    exec(mod.as_string(), {}, l)
  File "<string>", line 9, in <module>
KeyError: "'__name__' not in globals"
```

Testing earlier version of pylint-protobuf this error goes away so this is a regression. The regression looks to be introduced in 0.14. 0.13 checks the file properly, but 0.14 fails. The error message I gave was for 0.20.1 but the earlier versions have similar looking error messages.

My virtual environment pip freeze looks like,

```
absl-py==0.8.1
apache-beam==2.31.0
astroid==2.6.2
astunparse==1.6.3
avro-python3==1.9.2.1
cachetools==4.2.2
certifi==2021.5.30
cffi==1.14.6
charset-normalizer==2.0.3
crcmod==1.7
google-cloud-core==1.7.1
google-cloud-datastore==1.15.3
google-cloud-dlp==1.0.0
google-cloud-language==1.3.0
google-cloud-profiler==3.0.5
google-cloud-pubsub==1.7.0
google-cloud-spanner==1.19.1
google-cloud-videointelligence==1.16.1
google-cloud-vision==1.0.0
google-crc32c==1.1.2
google-pasta==0.2.0
google-resumable-media==1.3.1
googleapis-common-protos==1.53.0
grpc-google-iam-v1==0.12.3
grpcio==1.38.1
grpcio-gcp==0.2.2
h5py==2.10.0
hdfs==2.6.0
httplib2==0.19.1
idna==3.2
importlib-metadata==4.6.1
isort==5.9.2
Keras-Preprocessing==1.1.2
lazy-object-proxy==1.6.0
Markdown==3.3.4
mccabe==0.6.1
numpy==1.18.5
oauth2client==4.1.3
oauthlib==3.1.1
opt-einsum==3.3.0
packaging==21.0
pandas==1.3.0
proto-plus==1.19.0
protobuf==3.17.3
pyarrow==0.16.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.20
pydot==1.4.2
pylint==2.9.3
pylint-protobuf==0.20.1
pymongo==3.12.0
pyparsing==2.4.7
python-dateutil==2.8.2
pytz==2021.1
requests==2.26.0
requests-oauthlib==1.3.0
rsa==4.7.2
six==1.16.0
tensorboard==2.2.2
tensorboard-plugin-wit==1.8.0
tensorflow==2.2.3
tensorflow-estimator==2.2.0
tensorflow-metadata==0.22.2
tensorflow-serving-api==2.2.3
tensorflow-transform==0.22.0
termcolor==1.1.0
tfx-bsl==0.22.1
toml==0.10.2
typed-ast==1.4.3
typing-extensions==3.7.4.3
uritemplate==3.0.1
urllib3==1.26.6
Werkzeug==2.0.1
wrapt==1.12.1
zipp==3.5.0
```

All of those are dependencies of tensorflow_transform/pylint-protobuf and I just did pip install tensorflow-transform==0.22.0 && pip install pylint-protobuf==0.20.1 in a new virtual environment to get them. I'm on python 3.7.8

My .pylintrc file is minimal. It's just,

```
[MASTER]
load-plugins=pylint_protobuf 

[MESSAGES CONTROL]
# Only show errors and fatal.
disable=C, R, W
```

Any other information that would be helpful? Is this enough to reproduce the issue? I hope hardware version doesn't matter, but I am on Mac 11.4 in case it does. If that's not enough I can look into making docker image to reproduce the error.

**Comment by nelfin** [comment]
Hi @hmc-cs-mdrissi, thanks for the report. I'm not 100% sure of this, but it looks like that line is tripping up on the relative import in the Apache Beam compatibility layers since the transform is not checking the code in a package context. I never really liked the approach I took here. I'll add a guard to this to fix the issue you're seeing.

**Comment by nelfin** [comment]
@hmc-cs-mdrissi, try installing the latest pre-release with `pip install git+https://github.com/nelfin/pylint-protobuf@960a66303fdb0c840f1345f101fb7959ca5f9be2`

**Comment by hmc-cs-mdrissi** [comment]
I've tested it on my codebase and the plugin works now without errors. Thank you for the fix.

**Comment by nelfin** [comment]
Great to hear. This fix has gone out as part of a 0.20.2 release on PyPI.
